byzcoin/bcadmin/clicontracts/deferred.go

```diff
@@ -0,0 +1,374 @@
+package clicontracts
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/bcadmin/lib"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/protobuf"
+	"gopkg.in/urfave/cli.v1"
+)
+
+// DeferredSpawn is used to spawn a new deferred contract. It expects stdin to
+// contain the proposed transaction.
+func DeferredSpawn(c *cli.Context) error {
+	// Here is what this function does:
+	//   1. Parses the stdin in order to get the proposed transaction
+	//   2. Fires a spawn instruction for the deferred contract
+	//   3. Gets the response back
+
+	// ---
+	// 1.
+	// ---
+	proposedTransactionBuf, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		return errors.New("failed to read from stding: " + err.Error())
+	}
+
+	proposedTransaction := byzcoin.ClientTransaction{}
+	err = protobuf.Decode(proposedTransactionBuf, &proposedTransaction)
+	if err != nil {
+		return errors.New("failed to decode transaction: " + err.Error())
+	}
+
+	// ---
+	// 2.
+	// ---
+	bcArg := c.String("bc")
+	if bcArg == "" {
+		return errors.New("--bc flag is required")
+	}
+
+	cfg, cl, err := lib.LoadConfig(bcArg)
+	if err != nil {
+		return err
+	}
+
+	dstr := c.String("darc")
+	if dstr == "" {
+		dstr = cfg.AdminDarc.GetIdentityString()
+	}
+	d, err := lib.GetDarcByString(cl, dstr)
+	if err != nil {
+		return err
+	}
+
+	var signer *darc.Signer
+
+	sstr := c.String("sign")
+	if sstr == "" {
+		signer, err = lib.LoadKey(cfg.AdminIdentity)
+	} else {
+		signer, err = lib.LoadKeyFromString(sstr)
+	}
+	if err != nil {
+		return err
+	}
+
+	expireBlockIndexInt := uint64(6000)
```

Comment:
What's the reasoning behind 6000 ?

Reply:
For the moment it is an arbitrary high value because I have to specify an expireBlockIndex to the deferred contract, but don't want the client to specify it from the command line (the client should only have to add `--redirect` and nothing else to specify a proposed transaction). 
The plan is to remove this parameter here and make it an optional argument for the deferred contract, so that the deferred contract can use a default value when `expireBlockIndex` is not given (like `current_blockIdx + 100`).

Reply:
If #1875 is accepted before this one I can refactor this part.

Reply:
It's ok, you can do it in another PR